### PR TITLE
feat(Cart): Changes related to UI Cart

### PR DESCRIPTION
**Pull request by kibo-amolnadagonde**

### CODE REVIEW

File: components/AppHeader.vue [hunk 1]
```diff
@@ -24,18 +24,28 @@
...
+            @click="toggleCartSidebar"
+          >
             <SfIcon class="sf-header__icon" icon="empty_cart" size="1.25rem" />
+            <SfBadge v-if="totalItemsInCart" class="sf-badge--number sf-badge">{{
+              totalItemsInCart
+            }}</SfBadge>
           </SfButton>
         </div>
+        <div>
+          <button @click="addToCart({ product, quantity: 1 })">Add Product to Cart</button>
```
Comment (kibo-kevinwatts) [inline components/AppHeader.vue]: Remove test code

File: components/AppHeader.vue [hunk 2]
```diff
@@ -93,6 +105,18 @@ export default defineComponent({
 
     const isMobile = computed(() => mapMobileObserver().isMobile.get())
 
+    const { cart, load: loadCart, addToCart } = useCart()
+    // added temporary - to be removed after product page is developed
```
Comment (kibo-kevinwatts) [inline components/AppHeader.vue]: remove test code

File: components/AppHeader.vue [hunk 3]
```diff
@@ -112,6 +136,10 @@ export default defineComponent({
       unMapMobileObserver()
     })
 
+    onMounted(async () => {
```
Comment (kibo-kevinwatts) [inline components/AppHeader.vue]: useAsync

File: composables/getters/cartGetters.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  return {
+    regular: item.product?.price?.price,
+    special: item.product?.price?.salePrice,
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const getCartItemQty = (item: CartItem): number => item?.quantity
+
+export const getCartItemAttributes = (
```
Comment (kibo-kevinwatts) [inline composables/getters/cartGetters.ts:20]: For now, replace the logic inside this `getCartItemAttributes` with `item?.product?.options`

File: composables/getters/cartGetters.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+export const getCartItemSku = (item: CartItem) =>
+  item?.product?.sku || item?.product?.upc || item?.product?.productCode
+
+function getTotals(cart: Cart) {
+  return {
+    total: cart?.total,
+    subtotal: cart?.subtotal,
+    special: [cart?.orderDiscounts]?.length > 0 ? cart?.discountedSubtotal : cart?.subtotal,
```
Comment (kibo-kevinwatts) [inline composables/getters/cartGetters.ts]: remove [ ] around cart?.orderDiscounts?.length

File: components/CartSidebar.vue [hunk 6]
```diff
@@ -0,0 +1,257 @@
+<template>
```
Comment (kibo-kevinwatts) [inline components/CartSidebar.vue]: Remove this component

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: composables/getters/cartGetters.ts [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+export const getCartItemPrice = (
+  item: CartItem
+): { regular: Maybe<number> | undefined; special: Maybe<number> | undefined } => {
+  return {
+    regular: item.product?.price?.price,
+    special: item.product?.price?.salePrice,
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (kibo-sushant) [inline composables/getters/cartGetters.ts]: Remove all // eslint-disable-next-line @typescript-eslint/no-unused-vars

File: composables/useCart.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
+import { useNuxtApp } from "#app"
+import { getCartQuery } from "~~/gql/queries/cart"
+import { addToCartMutation } from "~~/gql/mutations"
+
+export const useCart = () => {
+  const cart = ref({})
```
Comment (kibo-kevinwatts) [inline composables/useCart.ts]: Refs in useCart need to be changed to useState to share the same cart across components

File: layouts/default.vue [hunk 9]
```diff
@@ -6,7 +6,7 @@
     <div class="content">
       <Nuxt />
     </div>
-
+    <Nuxt />
```
Comment (kibo-kevinwatts) [inline layouts/default.vue]: remove this

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: components/AppHeader.vue (hunk 1 above)
Comment (kibo-geetanshuchhabra) [reply components/AppHeader.vue]: Resolved

File: components/AppHeader.vue (hunk 2 above)
Comment (kibo-geetanshuchhabra) [reply components/AppHeader.vue]: Resolved

File: components/AppHeader.vue (hunk 3 above)
Comment (kibo-geetanshuchhabra) [reply components/AppHeader.vue]: Resolved

File: composables/getters/cartGetters.ts (hunk 4 above)
Comment (kibo-geetanshuchhabra) [reply composables/getters/cartGetters.ts:20]: Resolved

File: composables/getters/cartGetters.ts (hunk 5 above)
Comment (kibo-geetanshuchhabra) [reply composables/getters/cartGetters.ts]: Resolved

File: layouts/default.vue (hunk 9 above)
Comment (kibo-geetanshuchhabra) [reply layouts/default.vue]: Resolved

File: composables/useCart.ts (hunk 8 above)
Comment (kibo-geetanshuchhabra) [reply composables/useCart.ts]: Resolved

File: composables/getters/cartGetters.ts (hunk 7 above)
Comment (kibo-geetanshuchhabra) [reply composables/getters/cartGetters.ts]: Resolved

File: components/CartSidebar.vue (hunk 6 above)
Comment (kibo-geetanshuchhabra) [reply components/CartSidebar.vue]: Resolved

**Comment by kibo-sushant** [comment]
Ass all issues are resolved, merging the code

**Pull request merged**